react-carousel-component/src/carousel.jsx

```diff
@@ -9,24 +9,24 @@ class Carousel extends React.Component {
     this.arrowClicks = this.arrowClicks.bind(this);
     this.renderProgressDots = this.renderProgressDots.bind(this);
     this.clickDots = this.clickDots.bind(this);
+    this.timerId = setInterval(this.updateCarousel, 3000);
   }
 
-  componentDidMount() {
-    setInterval(() => {
-      if (this.state.index === this.props.images.length - 1) {
-        this.setState({
-          index: 0
-        });
-      } else {
-        this.setState({
-          index: this.state.index + 1
-        });
-      }
-    }, 3000);
+  updateCarousel() {
+    if (this.state.index === this.props.images.length - 1) {
+      this.setState({
+        index: 0
+      });
+    } else {
+      this.setState({
+        index: this.state.index + 1
+      });
+    }
   }
 
   arrowClicks(event) {
     const index = this.state.index;
+    this.componentDidMount();
```

Comment:
It's generally not good to call `componentDidMount` or any other lifecycle methods (except `setState`) yourself.